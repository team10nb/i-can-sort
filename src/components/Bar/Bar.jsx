import React from 'react';

class Bar extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            color: this.props.color,
            num: this.props.num,
        };
    }

    render(){
        const color = this.state.color;
        const num = this.state.num;
        // height和num之间可能要做一些换算之类的，先这样写了
        const height = num;
        
        let barStyle = {
            height: height + 'px', 
            width: '30px',
            background: color,
            // background: 'red',
            borderWidth: '10px',
            borderRadius: '8px',
        };

        const fontStyle = {
            margin: '0px auto',
            padding: '0px',
            textAlign: 'center',
            fontSize: '20px',
            fontColor: 'black',
            letterSpacing: '0px',
        }

        
        return (
            <div style = {barStyle}>
                <p style = {fontStyle}>{num}</p>
            </div>
        )
    }
}


export default Bar;