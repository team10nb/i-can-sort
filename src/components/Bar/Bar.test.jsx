import { render } from "@testing-library/react";
import Bar from "./Bar";

it('should match snapshot', () => {
    const color = 'red';
    const num = 90;
    const bar = render(<Bar color={color} num={num}/>);
    expect(bar).toMatchSnapshot();
})