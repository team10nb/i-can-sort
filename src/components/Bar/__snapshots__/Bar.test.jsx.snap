// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="height: 90px; width: 30px; background: red; border-width: 10px; border-radius: 8px;"
      >
        <p
          style="margin: 0px auto; padding: 0px; text-align: center; font-size: 20px; letter-spacing: 0px;"
        >
          90
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="height: 90px; width: 30px; background: red; border-width: 10px; border-radius: 8px;"
    >
      <p
        style="margin: 0px auto; padding: 0px; text-align: center; font-size: 20px; letter-spacing: 0px;"
      >
        90
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
